jina/drivers/reduce.py

```diff
@@ -31,12 +30,13 @@ def __call__(self, *args, **kwargs):
         self._traverse_apply(self.docs, *args, **kwargs)
         self.doc_pointers.clear()
 
-    def _apply_root(self, docs: 'DocumentSet', context_doc: 'Request', field: str, *args, **kwargs):
+    def _apply_root(self, docs: 'DocumentSet', field: str, *args, **kwargs):
         docs = []
         for doc in self.docs:
             docs.append(doc)
-        context_doc.body.ClearField(field)
-        getattr(context_doc, field).extend(docs)
+        request = self.msg.request
+        request.body.ClearField(field)
+        request.docs.extend(docs)
```

Comment:
does it work for u? I had some complaints about generator

Reply:
Your newly added test works. I am now waiting for CI results.

Reply:
Oh shit rebasing was not the smartest idea...